File: src/main/java/org/jukeboxmc/event/player/PlayerToggleFlyEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.event.Cancellable;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerToggleFlyEvent extends PlayerEvent implements Cancellable {

    private final boolean isFlying;

    /**
     * Creates a new {@link PlayerEvent}
     *
     * @param player who represents the player which comes with this event
     */
    public PlayerToggleFlyEvent( Player player, boolean isFlying ) {
        super( player );
        this.isFlying = isFlying;
    }

    public boolean isFlying() {
        return this.isFlying;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerToggleFlyEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.event.Cancellable
import org.jukeboxmc.api.player.Player

class PlayerToggleFlyEvent(
    player: Player,
    private val flying: Boolean
): PlayerEvent(player), Cancellable {

    fun isFlying(): Boolean {
        return this.flying
    }

}